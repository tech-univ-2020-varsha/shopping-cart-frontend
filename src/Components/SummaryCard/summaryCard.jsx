import React from 'react';
import propTypes from 'prop-types';
import * as styles from './summaryCard.module.css';

const SummaryCard = ({ cartTotal }) => (
  <div className={styles.container}>
    <div className={styles.summaryDetails}>
      <div className={styles.priceDetails}>
        <div className={styles.totalWord}>

          Total
        </div>
        <div className={styles.priceCartTotal}>
          <span className={styles.RS}>RS</span>
          {' '}
          {cartTotal}
        </div>
      </div>
      <hr className={styles.line} />
      <div className={styles.checkout}>
        <button type="submit" className={styles.checkoutBtn}>
          Checkout
        </button>
      </div>
    </div>
    <div>
      <button type="submit">
        Continue Shopping
      </button>
    </div>
  </div>
);

SummaryCard.propTypes = {
  cartTotal: propTypes.number.isRequired,
};
export default SummaryCard;
