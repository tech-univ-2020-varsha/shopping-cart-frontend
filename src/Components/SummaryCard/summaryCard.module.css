.container{
    display: flex;
    margin-left: 3em;
    width: 100%;
    flex-direction: column;
}

.summaryDetails{
    display: flex;
    flex-direction: column;
    background: rgb(196,196,196);
    padding: 2em;
    border-radius: 10px;
    width: 100%;
    margin-top: 2em;
    box-shadow: 0px 0px 6px #888888;
}

.priceDetails{
    display: flex;
   justify-content: space-between;
  font-size: 2em;
}


.continue{
    width: 100%;
    margin-top: 2em; 
}
.priceCartTotal{
display: flex;
align-self: end;
right:0;
position: relative;

}

.totalWord{
    display: flex;
    align-self: start;
}

.RS{
    font-size: 0.4em;
    position: absolute;
    left: 0;
    bottom: 0;
    transform: translate(-100%,0%);;
}

.line{
    margin-top: 2em;
     border: 0.5px solid black;
    width: 100%;
    background: black;
}

.checkout{
    width: 100%;
    margin-top: 2em;
    border: none;
}

.checkoutBtn{
    width: 100%;
    padding-left: 2em;
    padding-top: 1em;
    padding-right: 2em;
    padding-bottom: 1em;
    border-radius: 10px;
    background: rgb(238,195,92);
    box-shadow: 0px 0px 6px #888888;
    font-size: 1em;
    outline: none;
    margin-top: 2em;
    border: none;
}

.shoppingBtn{
    width: 100%;
    border-radius: 10px;
    background: rgb(238,195,92);
    box-shadow: 0px 0px 6px #888888;
    font-size: 1em;
    outline: none;
    border: none;
   
}




.continueBtn{
    width: 100%;
    border-radius: 10px;
    box-shadow: 1px 3px 5px #888888;
    font-size: 1em;
    outline: none;
    margin-top: 2em;
    padding: 1em 3em;
}


