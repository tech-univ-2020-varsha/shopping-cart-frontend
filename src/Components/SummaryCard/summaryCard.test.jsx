import { render } from '@testing-library/react';
import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import SummaryCard from './summaryCard';

describe('The Summary Card component', () => {
  const history = createMemoryHistory();
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router history={history}>
        <SummaryCard
          cartTotal={1}
          setTotal={() => {}}
        />

      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
