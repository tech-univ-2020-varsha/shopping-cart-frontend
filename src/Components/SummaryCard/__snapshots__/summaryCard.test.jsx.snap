// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Summary Card component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          Total
        </div>
        <div>
          <span>
            RS
          </span>
           1
        </div>
      </div>
      <hr />
      <div>
        <button
          type="submit"
        >
          Checkout
        </button>
      </div>
    </div>
    <div>
      <button
        type="submit"
      >
        Continue Shopping
      </button>
    </div>
  </div>
</DocumentFragment>
`;
